tests/curve_fit_test.py

```diff
@@ -0,0 +1,507 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from functools import partial
+
+import numpy as np
+import pytest
+
+import scipp as sc
+from scipp import curve_fit
+from scipp.compat.xarray_compat import from_xarray, to_xarray
+
+
+def func(x, a, b):
+    return a * sc.exp(-b * x)
+
+
+def func2d(x, t, a, b):
+    return a * sc.exp(-b * t / (1 + x))
+
+
+def func3d(x, t, y, a, b):
+    return a * y * sc.exp(-b * t / (1 + x))
+
+
+def func_np(x, a, b):
+    return a * np.exp(-b * x)
+
+
+def func2d_np(x, t, a, b):
+    return a * np.exp(-b * t / (1 + x))
+
+
+def func3d_np(x, t, y, a, b):
+    return a * y * np.exp(-b * t / (1 + x))
+
+
+def array(coords, f, params, noise_scale, seed=1234):
+    rng = np.random.default_rng(seed)
+    da_coords = {
+        c: sc.linspace(
+            dim=kw.pop('dim', c),
+            **kw,
+        )
+        for c, kw in coords.items()
+    }
+    data = f(**da_coords, **params)
+    # Noise is random but avoiding unbounded values to avoid flaky tests
+    data.values += noise_scale * np.clip(rng.normal(size=data.values.shape), -1.5, 1.5)
+    return sc.DataArray(data, coords=da_coords)
+
+
+def array1d(*, a=1.2, b=1.3, noise_scale=0.1, size=50):
+    return array(
+        dict(x=dict(dim='xx', start=-0.1, stop=4.0, num=size)),
```

Comment:
Why is the coord name not matching the dim?

Reply:
Because in general they don't have the same name, and I want to make sure the tests pass in the general case and not only in the special case when they have the same name.

Reply:
Maybe make it more explicitly different then. As it is now it looks like a typo that a future editor might "fix".